src/Api/Encoder.php

```diff
@@ -101,35 +108,68 @@ public function run(ImageInterface $image): EncodedImageInterface
      */
     public function getFormat(ImageInterface $image): string
     {
-        $fm = (string) $this->getParam('fm');
+        try {
+            $mediaType = $this->getMediaType($image);
+            $extensions = $mediaType->format()->fileExtensions();
 
-        if ($fm && array_key_exists($fm, static::supportedFormats())) {
-            return $fm;
+            return reset($extensions)->value; // @phpstan-ignore-line
+        } catch (\Exception) {
+            return 'jpg';
         }
-
-        /** @psalm-suppress RiskyTruthyFalsyComparison */
-        return array_search($image->origin()->mediaType(), static::supportedFormats(), true) ?: 'jpg';
     }
 
     /**
      * Get a list of supported image formats and MIME types.
      *
-     * @return array<string,string>
+     * @return array<string,MediaType>
```

Comment:
Changing the type can cause problems for users upgrading from 2.x. I prefer not breaking backwards compatibility unless absolutely necessary.

Reply:
fixed